src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -363,20 +363,24 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 		this._onDidRegisterSupportedExecutions.fire();
 	}
 
+	private async _runAutomaticTasks(): Promise<void> {
+		this.getWorkspaceTasks().then(async (tasks) => {
+			RunAutomaticTasks.runWithPermission(this, this._storageService, this._notificationService, this._workspaceTrustManagementService, this._openerService, this._configurationService, tasks);
+		});
+	}
+
 	private _attemptTaskReconnection(): void {
-		if (this._lifecycleService.startupKind !== StartupKind.ReloadedWindow) {
-			this._tasksReconnected = true;
+		if (this._tasksReconnected
+			|| !this._configurationService.getValue(TaskSettingId.Reconnection)
+			|| this._lifecycleService.startupKind !== StartupKind.ReloadedWindow) {
 			this._storageService.remove(AbstractTaskService.PersistentTasks_Key, StorageScope.WORKSPACE);
-		}
-		if (!this._configurationService.getValue(TaskSettingId.Reconnection) || this._tasksReconnected) {
 			this._tasksReconnected = true;
-			return;
+			this._runAutomaticTasks().then(() => { return; });
```

Comment:
Why do we need to run automatic tasks here? Does the `RunAutomaticTask` workbench contribution not work?

Reply:
It does, but there have been issues filed because trust is requested immediately when the workspace opens - regardless of whether there are automatic tasks.

Reply:
Also, shouldn't this only need to be run once? Perhaps a change I've made is responsible, but looks like it would run any time a task gets run.